include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -85,48 +161,82 @@ class sycl_backend
     template <typename NativeUniverseVector>
     sycl_backend(const NativeUniverseVector& v)
     {
+        bool profiling = true;
         global_rank_.reserve(v.size());
         for (auto e : v)
         {
             global_rank_.push_back(e);
+            if (!e.template has_property<sycl::property::queue::enable_profiling>())
+            {
+                profiling = false;
+            }
         }
+        is_profiling_enabled = profiling;
         sgroup_ptr_ = std::make_unique<submission_group>(global_rank_);
     }
 
     template <typename SelectionHandle, typename Function, typename... Args>
     auto
     submit(SelectionHandle s, Function&& f, Args&&... args)
```

Comment:
@AnuyaWelling2801 I think we are able to make this function a little bit shorter: please take a loot at PR https://github.com/oneapi-src/oneDPL/pull/1547

Reply:
Merged